res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,718 @@
+    engine.setValue(group, "rate", -script.pitch(control, value, status));
+};
+
+// Trax navigation functions
+// ========================================================
+ReloopBeatmix24.traxSelect = function(value, step) {
+    switch (traxMode) {
+        case 1: // Playlist mode
+            var i, j;
+            for (i = 0; i < Math.abs(value); i++) {
+                for (j = 0; j < step; j++) {
+                    if (value < 0) {
+                        engine.setValue("[Playlist]", "SelectPrevPlaylist",
+                            true);
+                    } else {
+                        engine.setValue("[Playlist]", "SelectNextPlaylist",
+                            true);
+                    }
+                }
+            }
+            break;
+        case 2: // Track mode
+            engine.setValue("[Playlist]", "SelectTrackKnob", value * step);
+            break;
+        case 3: // Preview mode
+            engine.setValue("[PreviewDeck1]", "playposition", Math.max(0,
+                Math.min(1, engine.getValue("[PreviewDeck1]",
+                    "playposition") + 0.02 * value * step)));
+            break;
+    }
+};
+
+ReloopBeatmix24.TraxTurn = function(channel, control, value, status, group) {
+    ReloopBeatmix24.traxSelect(value - 0x40, 1);
+};
+
+ReloopBeatmix24.ShiftTraxTurn = function(channel, control, value, status, group) {
+    ReloopBeatmix24.traxSelect(value - 0x40, 10);
+};
+
+ReloopBeatmix24.TraxPush = function(channel, control, value, status, group) {
+    switch (traxMode) {
+        case 1: // Playlist mode
+            engine.setValue("[Playlist]", "ToggleSelectedSidebarItem",
+                value);
+            break;
+        case 2: // Track mode
+            engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay",
+                value);
+            traxMode = 3;
+            break;
+        case 3: // Preview mode
+            if (value == DOWN) {
+                engine.setValue("[PreviewDeck1]", "play",
+                    engine.getValue("[PreviewDeck1]", "play") ? 0 : 1);
+            }
+            break;
+    }
+};
+
+ReloopBeatmix24.ShiftTraxPush = function(channel, control, value, status, group) {
+    ReloopBeatmix24.TraxPush(channel, control, value, status, group);
+};
+
+ReloopBeatmix24.BackButton = function(channel, control, value, status, group) {
+    if (value === DOWN) {
+        switch (traxMode) {
+            case 1: // Playlist mode
+                traxMode = 2; // Switch to track mode
+                break;
+            case 2: // Track mode
+                traxMode = 1; // Switch to playlist mode
+                break;
+            case 3: // Preview mode
+                traxMode = 2; // Switch to track mode
+                break;
+        }
+    }
+};
+
+ReloopBeatmix24.LoadButtonEject = function(group) {
+    loadButtonLongPressed[group] = true;
+    engine.setValue(group, "eject", 1);
```

Comment:
I added a note to the wiki (http://mixxx.org/wiki/doku.php/mixxxcontrols ) to advise script authors to do this.